.github/workflows/ci.yml

```
@@ -123,3 +123,16 @@ jobs:
         with:
           name: tmpnet-data
           path: ~/.tmpnet/networks/1000
+  mock_gen:
+    name: MockGen Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.min_go_version }}
+          check-latest: true
+      - shell: bash
+        run: scripts/mock.gen.sh
+      - shell: bash
+        run: .github/workflows/check-clean-branch.sh
```

Why not directly inline the script content in there + remove .github/workflows/check-clean-branch.sh (one less file is always a win 😄)

```suggestion
        run: |
            git update-index --really-refresh >> /dev/null
            git diff --exit-code
```

Also do we really need `git update-index --really-refresh >> /dev/null`? I never had the use case, I think `git diff --exit-code` should suffice to find diffs and exit with 1 if any diff is found.